### ISSUE TITLE
AVT 3 - React Combobox Component iOS screenreader issues

### ISSUE THREAD

**Issue by snidersd** [carbon-design-system/carbon#3418]
## Environment
iOS 12.3.1 with VoiceOver and Safari
## Description
1. Swipe Right to the title, VO announces "Combobox tiltle main"
2. Swipe Right to the optional text, VO announces "optional helper text here'
3. Swipe Right to the Combobox, VO announces  that its a pop-up menu and double tap to expand.
**Expected Result:** VO should also announce the text "Filter..." 
4. Double tap, VO announces that there are 5 results and to use the arrow keys to navigate.
5. Swipe Right, to move down the list and Swipe Left to move up the list. VO also announces the beginning and end of the list. 
6. Double tap to select Option 1, VO announces "close collapsed, option 1, option 1.
Note: option 1 should only be announced once.  
7. Swipe right and focus skips the X icon button and moves focus away from the Combobox.
**Expected Result:** Expected the button to receive focus and the button purpose announced.
8. Shift-Tab back to the Combobox and VO announces "Open menu pop-up double tap to expand."
**Expected Result:** VO should have also announced Option 1 that was previously  selected in step 6, which is still visibly selected in the Combobox.

**Comment by abbeyhrt** [comment]
Closing since it seems to be a duplicate of #3419

**Comment by snidersd** [comment]
@abbeyhrt This is not a duplicate of #3419. This issue is related to a failure on an iPhone using iOS and #3418 is a failure on macOS. The navigation for VoiceOver testing is very different.

**Comment by abbeyhrt** [comment]
@snidersd ah okay, sorry about that!

**Comment by jnm2377** [comment]
1. ✅ Swipe Right to the title, VO announces "Combobox title main"
2. ✅ Swipe Right to the optional text, VO announces "optional helper text here'
3. ✅ Swipe Right to the Combobox, VO announces  that its a pop-up menu and double tap to expand.
- [x] **Expected Result:** VO should also announce the text "Filter..." 
4. ✅ Double tap, VO announces that there are 5 results and to use the arrow keys to navigate.
5. ✅ Swipe Right, to move down the list and Swipe Left to move up the list. VO also announces the beginning and end of the list. 
6. ✅ Double tap to select Option 1, VO announces "close collapsed, option 1, option 1.
- [x] Note: option 1 should only be announced once.  
7. ✅ Swipe right and focus skips the X icon button and moves focus away from the Combobox.
- [x] **Expected Result:** Expected the button to receive focus and the button purpose announced.
8. ✅ Shift-Tab back to the Combobox and VO announces "Open menu pop-up double tap to expand."
- [x] **Expected Result:** VO should have also announced Option 1 that was previously  selected in step 6, which is still visibly selected in the Combobox.

@snidersd I just tested our latest react on my iPhone with VoiceOver doing exactly what you said to do, and it looks like the voice over changes you were requesting are working as expected now.

- Filter is being announced when no item is selected
- When an option is selected, it is only announced once
- The clear button is not being skipped and it says `clear selection`
- When I go back to the combobox after focusing on the clear selection button, it is announcing the previously selected option
